cnd/src/db/save.rs

```diff
@@ -516,3 +518,84 @@ impl Save<CreatedSwap<han::CreatedSwap, halight::CreatedSwap>> for Sqlite {
         unimplemented!()
     }
 }
+
+#[async_trait]
+impl Save<CreatedSwap<herc20::CreatedSwap, halight::CreatedSwap>> for Sqlite {
+    async fn save(
+        &self,
+        created: CreatedSwap<herc20::CreatedSwap, halight::CreatedSwap>,
+    ) -> anyhow::Result<()> {
+        let local_swap_id = created.swap_id;
+        let role = created.role;
+
+        // Save the swap into the swaps table.
+        let swap = ISwap::new(created.swap_id, created.peer, created.role);
+        self.save_swap(&swap).await?;
+
+        // Save tha address hint if present
+        if let Some(address_hint) = created.address_hint {
+            self.save_address_hint(local_swap_id, address_hint).await?;
+        }
+
+        // Save the herc20 details.
+        let redeem_identity = match role {
+            Role::Alice => None,
+            Role::Bob => Some(Text(EthereumAddress::from(created.alpha.identity))),
+        };
+        let refund_identity = match role {
+            Role::Alice => Some(Text(EthereumAddress::from(created.alpha.identity))),
+            Role::Bob => None,
+        };
+        let herc = IHerc20SwapDetail {
+            swap_id: 0, // FK, set during save.
+            amount: Text(created.alpha.amount.into()),
+            chain_id: U32(created.alpha.chain_id),
+            expiry: U32(created.alpha.absolute_expiry),
+            hash_function: Text(HashFunction::Sha256),
+            token_contract: Text(created.alpha.contract_address.into()),
+            redeem_identity,
+            refund_identity,
+            ledger: "alpha".to_string(),
+        };
+
+        self.save_herc20_swap_detail(local_swap_id, &herc).await?;
+
+        // Save the halight details.
+        let redeem_identity = match role {
+            Role::Alice => Some(Text(created.beta.identity.into())),
+            Role::Bob => None,
+        };
+        let refund_identity = match role {
+            Role::Alice => None,
+            Role::Bob => Some(Text(created.beta.identity.into())),
+        };
+        let network = BitcoinNetwork::from_str(&created.beta.network)?;
+        let halight = IHalightSwapDetail {
+            swap_id: 0, // FK, set during save.
+            amount: Text(created.beta.amount.into()),
+            network: Text(network),
+            chain: "bitcoin".to_string(), // FIXME: What should this be?
+            expiry: U32(0),               // FIXME: What should this be?
```

Comment:
I think we can remove this. We added `expiry` and `cltv_expiry` because they co-existed at the time when we wrote the spike. 
`expiry` was since removed as far as I could see from the code, so I'd remove this unless @thomaseizinger @bonomat @D4nte disagree.

(same for the `chain` parameter...?)

Reply:
My understanding was that we need both: 
* `expiry` is for the invoice
* `cltv_expiry` is the actual payment